prime-dotnet-webapi/Services/SiteService.cs

```diff
@@ -282,6 +287,11 @@ public async Task<Site> UpdatePecCode(int siteId, string pecCode)
             var site = await GetBaseSiteQuery()
                 .SingleOrDefaultAsync(s => s.Id == siteId);
 
+            if (await IsNonHaSiteAndPecNotUnique(site, pecCode))
+            {
+                throw new InvalidOperationException($"Could not update the site.");
```

Comment:
We should not be throwing an error in a service method like this due to business logic; especially if the controller is not validating this information. If you try to update the PEC by calling PUT /sites/5/pec and the PEC happens to exist the server will respond with a `500 Internal Service Error` due to this uncaught exception. The controller should be validating this at a minimum.